### PR TITLE
Add script hash mock test

### PR DESCRIPTION
**Pull request by usatie**
- Fix some issues related to P2SH
- Add test cases of raw/P2SH multi-sig script mock

### CODE REVIEW

File: BitcoinCashKit/Core/Scripts/ScriptExecutionContext.swift [hunk 1]
```diff
@@ -74,6 +74,7 @@ public class ScriptExecutionContext {
     }
 
     public func shouldVerifyP2SH() -> Bool {
+        print(NSTimeIntervalSince1970, blockTimeStamp, BTC_BIP16_TIMESTAMP)
```
Comment (akifuj) [inline BitcoinCashKit/Core/Scripts/ScriptExecutionContext.swift]: Is `Date().timeIntervalSince1970` better than `NSTimeIntervalSince1970 `?

File: BitcoinCashKitTests/MockHelperTests.swift [hunk 2]
```diff
@@ -26,49 +26,173 @@ import XCTest
...
-            let result = try verifyCustomMultisig(with: key)
-            XCTAssertFalse(result, "\(key) Should fail but succeeds.")
-        } catch OpCodeExecutionError.error("OP_EQUALVERIFY failed.") {
-            // Expected fail:  do nothing
-        } catch let error {
-            XCTFail("Inappropriate error for \(key) : \(error)")
+    // custom multisig unlock
+    struct CustomMultisigUnlockScriptBuilder: SingleKeyScriptBuilder {
+        func build(with sigWithHashType: Data, key: MockKey) -> Script {
+            if key == MockKey.keyA {
```
Comment (akifuj) [inline BitcoinCashKitTests/MockHelperTests.swift]: Is `switch` sentence better, ins't it?

File: BitcoinCashKit/Core/Scripts/ScriptExecutionContext.swift (hunk 1 above)
Comment (usatie) [reply BitcoinCashKit/Core/Scripts/ScriptExecutionContext.swift]: They are different.

File: BitcoinCashKitTests/MockHelperTests.swift (hunk 2 above)
Comment (usatie) [reply BitcoinCashKitTests/MockHelperTests.swift]: That's true.

File: BitcoinCashKit/Core/Scripts/ScriptExecutionContext.swift (hunk 1 above)
Comment (akifuj) [reply BitcoinCashKit/Core/Scripts/ScriptExecutionContext.swift]: Okay!

**Pull request merged**